tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -400,6 +400,103 @@ TILEDB_EXPORT int32_t tiledb_query_get_status_details(
     tiledb_query_t* query,
     tiledb_query_status_details_t* status_details) TILEDB_NOEXCEPT;
 
+/* ****************************** */
+/*          QUERY CONDITION       */
+/* ****************************** */
+
+/**
+ * Allocates a TileDB query condition value object.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * uint32_t value = 5;
+ * tiledb_query_condition_t* query_condition;
+ * tiledb_query_condition_alloc_value(
+ *    ctx,
+ *    "longitude",
+ *    &value,
+ *    sizeof(value),
+ *    TILEDB_LT,
+ *    &query_condition);
+ *
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param attribute_name The attribute name.
+ * @param condition_value The value to compare against an attribute value.
+ * @param condition_value_size The byte size of `condition_value`.
+ * @param op The comparison operator.
+ * @param query_condition The query condition to create
+ * @return `TILEDB_OK` for success and `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_query_condition_alloc_value(
+    tiledb_ctx_t* ctx,
+    const char* attribute_name,
+    const void* condition_value,
+    uint64_t condition_value_size,
+    tiledb_query_condition_op_t op,
+    tiledb_query_condition_t** cond) TILEDB_NOEXCEPT;
+
+/**
+ * Allocates a TileDB query condition expression object.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * uint32_t value_1 = 5;
+ * tiledb_query_condition_t* query_condition_1;
+ * tiledb_query_condition_alloc(
+ *   ctx,
+ *   "longitude",
+ *   &value_1,
+ *   sizeof(value_1),
+ *   TILEDB_LT,
+ *   &query_condition_1);
+ *
+ * uint32_t value_2 = 20;
+ * tiledb_query_condition_t* query_condition_2;
+ * tiledb_query_condition_alloc_value(
+ *   ctx,
+ *   "latitude",
+ *   &value_2,
+ *   sizeof(value_2),
+ *   TILEDB_GE,
+ *   &query_condition_2);
+ *
+ * tiledb_query_condition_t* conditions[2];
+ * conditions[0] = &query_condition_1;
+ * conditions[1] = &query_condition_2;
```

Comment:
This example would be clearer with an array of three conditions. We already have support for binary construction.

Reply:
Fixed.